src/ast/numeric-literal.ts

```diff
@@ -0,0 +1,14 @@
+import Context from '../semantics/context';
+import { IAstNode } from '../type-definitions/plainscript-types';
+
+export default class NumericLiteral implements IAstNode<NumericLiteral> {
+  constructor(public value: number) { }
+  // Intentionally empty.
+  // tslint:disable-next-line:no-empty
+  public analyze(_: Context) { }
+  public optimize(): NumericLiteral { return this; }
+
+  // Depends on the target language, thus gets filled in
+  // by the necessary generator at runtime.
+  public gen() { }
```

Comment:
So I'm starting to wonder....Is there any way you can use [Mixins](https://www.typescriptlang.org/docs/handbook/mixins.html_ to remove all these `gen` declarations?

Reply:
Oh, I'll give it a shot !